crates/ty_server/src/server/api/requests/semantic_tokens.rs

```diff
@@ -33,14 +36,26 @@ impl BackgroundDocumentRequestHandler for SemanticTokensRequestHandler {
             return Ok(None);
         }
 
-        let Some(file) = snapshot.to_file(db) else {
+        let Some(file) = snapshot.to_notebook_or_file(db) else {
             return Ok(None);
         };
 
+        let mut range = None;
```

Comment:
It might be worth renaming this to `cell_range` instead with a comment stating that this is required to limit the semantic token generation for the cell as `file` would represent an entire notebook.